### PR TITLE
JMK_31: Test cases changes updated of post mapping

### PR DESCRIPTION
**Pull request by sehajbirsingh1313**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -71,6 +71,7 @@
 			<version>1.18.36</version>
 			<scope>provided</scope>
 		</dependency>
+
```
Comment (pawanpreet193) [inline pom.xml]: blank line can be removed

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+        assertNull(certificateDTO.getCertificateId());
+        assertNull(certificateDTO.getCertificateName());
+        assertNull(certificateDTO.getIssuedBy());
+        assertNull(certificateDTO.getIssueDate());
+        assertNull(certificateDTO.getExpiryDate());
+        assertNull(certificateDTO.getCredentialUrl());
+        assertNull(certificateDTO.getUserId());
+    }
+
```
Comment (pawanpreet193) [inline src/test/java/com/jobmatrix/dto/CertificateDTOTest.java]: testing constructors, getters and setters may not be needed, as they are Lombok generated

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    @Test
+    void testBoundaryConditionsForDates() {
+        Date pastDate = new Date(System.currentTimeMillis() - 10L * 365 * 24 * 60 * 60 * 1000); // 10 years ago
+        Date futureDate = new Date(System.currentTimeMillis() + 10L * 365 * 24 * 60 * 60 * 1000); // 10 years ahead
```
Comment (pawanpreet193) [inline src/test/java/com/jobmatrix/dto/CertificateDTOTest.java]: may cause multiplication overflow ji,
`Date pastDate = new Date(System.currentTimeMillis() - TimeUnit.DAYS.toMillis(10 * 365)); // 10 years ago
Date futureDate = new Date(System.currentTimeMillis() + TimeUnit.DAYS.toMillis(10 * 365)); // 10 years ahead`
might be better

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    @Test
+    void testSameIssueAndExpiryDate() {
+        Date sameDate = new Date(System.currentTimeMillis());
+
+        certificateDTO.setIssueDate(sameDate);
+        certificateDTO.setExpiryDate(sameDate);
+
+        assertEquals(sameDate, certificateDTO.getIssueDate());
+        assertEquals(sameDate, certificateDTO.getExpiryDate());
+    }
```
Comment (pawanpreet193) [inline src/test/java/com/jobmatrix/dto/CertificateDTOTest.java]: bit doubtful about the utility of this test

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        UUID uuid1 = UUID.randomUUID();
+        UUID uuid2 = UUID.randomUUID();
+
+        certificateDTO.setUserId(uuid1);
+        assertEquals(uuid1, certificateDTO.getUserId());
+
+        certificateDTO.setUserId(uuid2);
+        assertEquals(uuid2, certificateDTO.getUserId());
+    }
+
```
Comment (pawanpreet193) [inline src/test/java/com/jobmatrix/dto/CertificateDTOTest.java]: may not be needed

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+        String specialString = "!@#$%^&*()_+";
+        certificateDTO.setCertificateName(specialString);
+        certificateDTO.setIssuedBy(specialString);
+        certificateDTO.setCredentialUrl(specialString);
+
+        assertEquals(specialString, certificateDTO.getCertificateName());
+        assertEquals(specialString, certificateDTO.getIssuedBy());
+        assertEquals(specialString, certificateDTO.getCredentialUrl());
+    }
+}
```
Comment (pawanpreet193) [inline src/test/java/com/jobmatrix/dto/CertificateDTOTest.java:126]: tests - testEmptyStrings, testLongStrings, testSpecialCharacters may not be needed

File: src/test/java/com/jobmatrix/dto/JobDTOTest.java [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+        // Assert
+        assertNull(fullJobDTO.getJobId());
+        assertNull(fullJobDTO.getJobTitle());
+        assertNull(fullJobDTO.getCompanyName());
+        assertNull(fullJobDTO.getStartDate());
+        assertNull(fullJobDTO.getEndDate());
+        assertNull(fullJobDTO.getJobResponsibilities());
+        assertNull(fullJobDTO.getUserId());
+    }
```
Comment (pawanpreet193) [inline src/test/java/com/jobmatrix/dto/JobDTOTest.java]: tests may not be needed for noArgsConctructor, AllArgsConstructor, getters and setters

File: src/test/java/com/jobmatrix/dto/JobDTOTest.java [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+
+        // Act
+        jobDTO.setStartDate(minDate);
+        jobDTO.setEndDate(maxDate);
+
+        // Assert
+        assertEquals(minDate, jobDTO.getStartDate());
+        assertEquals(maxDate, jobDTO.getEndDate());
+    }
+}
```
Comment (pawanpreet193) [inline src/test/java/com/jobmatrix/dto/JobDTOTest.java:178]: logic for testings boundary values for dates are missing

File: src/test/java/com/jobmatrix/serviceimpl/FreelancerProfileServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+        assertFalse(violations.isEmpty());
+        assertTrue(violations.stream().anyMatch(v -> v.getMessage().contains("Please enter a valid Phone Number")));
+    }
+
+    @Test
+    void testFreelancerDTO_DefaultValues() {
+        FreelancerDTO freelancerDTO = new FreelancerDTO();
+        assertFalse(freelancerDTO.isAbcMember(), "Default value of isAbcMember should be false.");
+    }
+}
```
Comment (pawanpreet193) [inline src/test/java/com/jobmatrix/serviceimpl/FreelancerProfileServiceImplTest.java:89]: most probably we need tests for only the create method here, freelancerDTO tests are already covered in the dto test

File: pom.xml (hunk 1 above)
Comment (sehajbirsingh1313) [reply pom.xml]: Done

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java (hunk 3 above)
Comment (sehajbirsingh1313) [reply src/test/java/com/jobmatrix/dto/CertificateDTOTest.java]: Done

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java (hunk 2 above)
Comment (sehajbirsingh1313) [reply src/test/java/com/jobmatrix/dto/CertificateDTOTest.java]: Done

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java (hunk 5 above)
Comment (sehajbirsingh1313) [reply src/test/java/com/jobmatrix/dto/CertificateDTOTest.java]: Done

File: src/test/java/com/jobmatrix/dto/JobDTOTest.java (hunk 7 above)
Comment (sehajbirsingh1313) [reply src/test/java/com/jobmatrix/dto/JobDTOTest.java]: Done

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java (hunk 4 above)
Comment (sehajbirsingh1313) [reply src/test/java/com/jobmatrix/dto/CertificateDTOTest.java]: removed

File: src/test/java/com/jobmatrix/dto/CertificateDTOTest.java (hunk 6 above)
Comment (sehajbirsingh1313) [reply src/test/java/com/jobmatrix/dto/CertificateDTOTest.java:126]: testSpecialCharacters may be required since the name of the file might contain the special characters. 

Other two removed

File: src/test/java/com/jobmatrix/dto/JobDTOTest.java (hunk 8 above)
Comment (sehajbirsingh1313) [reply src/test/java/com/jobmatrix/dto/JobDTOTest.java:178]: Done

File: src/test/java/com/jobmatrix/serviceimpl/FreelancerProfileServiceImplTest.java (hunk 9 above)
Comment (sehajbirsingh1313) [reply src/test/java/com/jobmatrix/serviceimpl/FreelancerProfileServiceImplTest.java:89]: test removed

File: src/test/java/com/jobmatrix/serviceimpl/FreelancerProfileServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+    @BeforeEach
+    void setUp() {
+        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+        validator = factory.getValidator();
+    }
+
+    @Test
+    void testValidFreelancerDTO() {
+        FreelancerDTO freelancerDTO = FreelancerDTO.builder()
```
Comment (ivesingh93) [inline src/test/java/com/jobmatrix/serviceimpl/FreelancerProfileServiceImplTest.java]: Why aren't we using createFreelancerDTO from the test factory for all the tests in this class?

**Review by pawanpreet193 — DISMISSED** [review]

File: src/test/java/com/jobmatrix/serviceimpl/FreelancerProfileServiceImplTest.java (hunk 10 above)
Comment (sehajbirsingh1313) [reply src/test/java/com/jobmatrix/serviceimpl/FreelancerProfileServiceImplTest.java]: Done

**Review by ivesingh93 — APPROVED** [review]

**Review by pawanpreet193 — APPROVED** [review]
